internal/sign/job/manager.go

```diff
@@ -5,40 +5,82 @@ import (
 	"errors"
 	"fmt"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
 	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/module"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/sign"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type signJobManager struct {
+	client    client.Client
 	signer    Signer
 	helper    sign.Helper
 	jobHelper utils.JobHelper
+	registry  registry.Registry
 }
 
-func NewSignJobManager(signer Signer, helper sign.Helper, jobHelper utils.JobHelper) *signJobManager {
+func NewSignJobManager(
+	client client.Client,
+	signer Signer,
+	helper sign.Helper,
+	jobHelper utils.JobHelper,
+	registry registry.Registry) *signJobManager {
 	return &signJobManager{
+		client:    client,
 		signer:    signer,
 		helper:    helper,
 		jobHelper: jobHelper,
+		registry:  registry,
+	}
+}
+
+func (jbm *signJobManager) ShouldSync(
+	ctx context.Context,
+	modSpec kmmv1beta1.ModuleSpec,
+	owner metav1.Object,
+	m kmmv1beta1.KernelMapping) (bool, error) {
+
+	// if there is no sign specified skip
+	if !module.ShouldBeSigned(modSpec, m) {
+		return false, nil
 	}
+
+	exists, err := module.ImageExists(ctx, jbm.client, jbm.registry, modSpec, owner.GetNamespace(), m, m.ContainerImage)
+	if err != nil {
+		return false, fmt.Errorf("failed to check existence of image %s: %w", m.ContainerImage, err)
+	}
+
+	return !exists, nil
 }
 
-func (jbm *signJobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string, imageToSign string, targetImage string, pushImage bool) (utils.Result, error) {
+func (jbm *signJobManager) Sync(
+	ctx context.Context,
+	modSpec kmmv1beta1.ModuleSpec,
+	owner metav1.Object,
+	m kmmv1beta1.KernelMapping,
+	targetKernel string,
+	imageToSign string,
+	targetImage string,
+	pushImage bool) (utils.Result, error) {
```

Comment:
How about adding variadic options to that function as well? Ultimately we will also have the TLS options here.